timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java

```diff
@@ -16,85 +16,109 @@
 import com.palantir.lock.RemoteLockService;
 import com.palantir.remoting2.config.ssl.SslSocketFactories;
-import com.palantir.timelock.TimeLockAgent;
+import com.palantir.timelock.Observables;
+import com.palantir.timelock.clock.ClockSkewMonitorCreator;
 import com.palantir.timelock.config.ImmutablePaxosRuntimeConfiguration;
 import com.palantir.timelock.config.ImmutableTimeLockDeprecatedConfiguration;
-import com.palantir.timelock.config.PaxosInstallConfiguration;
 import com.palantir.timelock.config.PaxosRuntimeConfiguration;
 import com.palantir.timelock.config.TimeLockDeprecatedConfiguration;
 import com.palantir.timelock.config.TimeLockInstallConfiguration;
 import com.palantir.timelock.config.TimeLockRuntimeConfiguration;
 
 import io.reactivex.Observable;
 
-public class PaxosAgent extends TimeLockAgent {
-    private final PaxosInstallConfiguration paxosInstall;
+public class TimeLockAgent {
+    private final TimeLockInstallConfiguration install;
+    private final Observable<TimeLockRuntimeConfiguration> runtime;
+    private final Consumer<Object> registrar;
 
     private final PaxosResource paxosResource;
     private final PaxosLeadershipCreator leadershipCreator;
     private final LockCreator lockCreator;
     private final PaxosTimestampCreator timestampCreator;
     private final TimeLockServicesCreator timelockCreator;
 
-    public PaxosAgent(TimeLockInstallConfiguration install,
+    public TimeLockAgent(TimeLockInstallConfiguration install,
             Observable<TimeLockRuntimeConfiguration> runtime,
             Consumer<Object> registrar) {
         this(install, runtime, ImmutableTimeLockDeprecatedConfiguration.builder().build(), registrar);
     }
 
-    public PaxosAgent(TimeLockInstallConfiguration install,
+    public TimeLockAgent(TimeLockInstallConfiguration install,
             Observable<TimeLockRuntimeConfiguration> runtime,
             TimeLockDeprecatedConfiguration deprecated,
             Consumer<Object> registrar) {
-        super(install, runtime, deprecated, registrar);
-        this.paxosInstall = install.algorithm();
+        this.install = install;
+        this.runtime = runtime;
+        this.registrar = registrar;
 
-        this.paxosResource = PaxosResource.create(paxosInstall.dataDirectory().toString());
+        this.paxosResource = PaxosResource.create(install.paxos().dataDirectory().toString());
         this.leadershipCreator = new PaxosLeadershipCreator(install, runtime, registrar);
         this.lockCreator = new LockCreator(runtime, deprecated);
         this.timestampCreator = new PaxosTimestampCreator(paxosResource,
                 PaxosRemotingUtils.getRemoteServerPaths(install),
                 PaxosRemotingUtils.getSslConfigurationOptional(install).map(SslSocketFactories::createSslSocketFactory),
-                runtime.map(PaxosAgent::getAlgorithmConfigOrDefault));
+                runtime.map(TimeLockAgent::getAlgorithmConfigOrDefault));
```

Comment:
Mm, good spot. Yeah, it should be. I'll check whether we should _rename_ or _remove_ it and update here